driver/src/contracts/stablex_contract.rs

```diff
@@ -42,6 +45,59 @@ pub trait StableXContract {
     ) -> Result<()>;
 }
 
+struct AuctionElement {
+    valid_from: U256,
+    valid_until: U256,
+    sell_token_balance: u128,
+    order: Order,
+}
+
+impl AuctionElement {
+    fn in_auction(&self, index: U256) -> bool {
+        self.valid_from < index && index <= self.valid_until
+    }
+
+    fn from_bytes(bytes: &[u8; 113]) -> Self {
+        let mut data_vector = bytes.to_vec();
+        let account_id = H160::pop_from_log_data(&mut data_vector);
+        let sell_token_balance = u128::pop_from_log_data(&mut data_vector);
+        let buy_token = u16::pop_from_log_data(&mut data_vector);
+        let sell_token = u16::pop_from_log_data(&mut data_vector);
+        let valid_from = U256::from(u32::pop_from_log_data(&mut data_vector));
+        let valid_until = U256::from(u32::pop_from_log_data(&mut data_vector));
+        let is_sell_order = bool::pop_from_log_data(&mut data_vector);
+        let price_numerator = u128::pop_from_log_data(&mut data_vector);
+        let price_denominator = u128::pop_from_log_data(&mut data_vector);
+        let remaining_amount = u128::pop_from_log_data(&mut data_vector);
+
+        // Todo - Will likely have to compute this differently.
+        // TODO - put in own function fn [u128; 3] -> [u128; 2]
+        let buy_amount: u128;
```

Comment:
I like the idea of putting this into its own method. Otherwise it'd be slightly more rusty to do

```rust
let (buy_amount, sell_amount) = if is_sell_order {
  (remaining_amount, (price_numerator * remaining_amount) / price_denominator)
} else {
  ((price_denominator * remaining_amount) / price_numerator, remaining_amount)
}
```

Reply:
I could also get behind the slightly more rusty idea (since I don't really want to have to name this function) and have been looking for a reason to use a `let if`.